ste/folderCreationTracker.go

```diff
@@ -23,8 +22,8 @@ func NewFolderCreationTracker(fpo common.FolderPropertyOption, plan *JobPartPlan
 		return &jpptFolderTracker{ // This prevents a dependency cycle. Reviewers: Are we OK with this? Can you think of a better way to do it?
 			plan:                   plan,
 			mu:                     &sync.Mutex{},
-			contents:               make(map[string]uint32),
-			unregisteredButCreated: make(map[string]struct{}),
+			contents:               common.NewTrie(),
+			unregisteredButCreated: common.NewTrie(),
```

Comment:
I'm also fine if we address this after the PR is merged but shouldnt both contents and unregisteredButCreated be able to be tracked in the same Trie? We can just add a boolean for unregisteredButCreated.

Reply:
I initially considered combining the data to optimize memory usage, but after further reflection, I realized that in terms of code readability, it's better to keep the data separate. By keeping the data distinct, the logic remains clearer, making it easier for developers to understand and maintain the code. 
Let me know your thoughts on this.

Reply:
Something of this sort right?

type TrieNode struct {
	Children               map[string]*TrieNode
	Value                  uint32
	unregisteredButCreated bool
	isEnd                  bool
}

Reply:
yep